File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/FcmContext.java
package com.jdroid.android.firebase.fcm;

import com.jdroid.java.collections.Lists;

import java.util.List;

public class FcmContext {

	private static List<FcmMessage> fcmMessages = Lists.newArrayList();

	private static List<FcmEventsListener> fcmEventsListeners = Lists.newArrayList();

	public static void addFcmMessage(FcmMessage fcmMessage) {
		fcmMessages.add(fcmMessage);
	}

	public static List<FcmMessage> getFcmMessages() {
		return fcmMessages;
	}

	public static List<FcmEventsListener> getFcmEventsListeners() {
		return fcmEventsListeners;
	}

	public static void addFcmEventsListener(FcmEventsListener fcmEventsListener) {
		fcmEventsListeners.add(fcmEventsListener);
	}
}

File: jdroid-android-firebase-fcm/src/main/java/com/jdroid/android/firebase/fcm/FcmContext.kt
package com.jdroid.android.firebase.fcm

object FcmContext {

    private val fcmMessages = mutableListOf<FcmMessage>()

    private val fcmEventsListeners = mutableListOf<FcmEventsListener>()

    fun addFcmMessage(fcmMessage: FcmMessage) {
        fcmMessages.add(fcmMessage)
    }

    fun getFcmMessages(): List<FcmMessage> {
        return fcmMessages
    }

    fun getFcmEventsListeners(): List<FcmEventsListener> {
        return fcmEventsListeners
    }

    fun addFcmEventsListener(fcmEventsListener: FcmEventsListener) {
        fcmEventsListeners.add(fcmEventsListener)
    }
}
